Update availability/platform badge

lshep opened the pull request:
- instead of convoluted combo of building and .bbsoption
    display reflection of .bbsoption and deprecation status

lshep committed:
Update availabilty/platform badge

  - instead of convoluted combo of building and .bbsoption
    display reflection of .bbsoption and deprecation status

mtmorgan commented on Rakefile:
```diff
@@ -448,29 +448,64 @@ end
-    json_file = File.join("assets", "packages", "json", numeric_version, "bioc",
-      "packages.json")
+    json_file = File.join("assets", "packages", "json", numeric_version, "bioc", "packages.json")
+    json_obj = JSON.parse(File.read(json_file))
```
for review clarity, I would have left L456, 7 where they were...

mtmorgan commented on Rakefile:
```diff
@@ -448,29 +448,64 @@ end
+              img = "some"
+            end
+          end
+          puts "Creating badge for #{pkg} :  #{img}"
```
Seems like this code chunk could be a function and reused a few lines down?

mtmorgan commented on Rakefile:
```diff
@@ -448,29 +448,64 @@ end
-	for item in meat_index
-	  view = json_obj[item['Package']]
-	  next if view.nil?
-	  get_available(item, numeric_version, view)
```
generally, I like this design better, where a function encapsulates an action. This makes it easier to think about isolated operations in the code, including testing the specific functionality without having to process all badges...

lshep committed:
Update colors for dependency badge

lshep committed:
Update to have helper functions

  - split into rake task and helper functions for easy of debugging
    and seperate functionality

lshep commented:
@mtmorgan   the code has been updated to separate out into helper functions.  Please re-review at earliest convenience

mtmorgan commented on Rakefile:
```diff
@@ -454,49 +454,27 @@ task :get_availability_shields  do
+	end
       else
-        puts "ERROR meat_index nil"
+	puts "ERROR meat_index nil"
```
sorry to be a pain but would rather not see the whitespace changes here and below...

lshep replied:
how do I change back to a whitespace-cleanup call without losing the changes?
